ffi/harfbuzz.lua

```diff
@@ -1,3 +1,118 @@
+local coverage = require("ffi/harfbuzz_coverage")
 local ffi = require("ffi")
+local hb = ffi.load("libs/libharfbuzz." .. (ffi.os == "OSX" and "0.dylib" or "so.0"))
+local HB = setmetatable({}, {__index = hb})
+
 require("ffi/harfbuzz_h")
-return ffi.load("libs/libharfbuzz." .. (ffi.os == "OSX" and "0.dylib" or "so.0"))
+
+local hb_face_t = {}
+hb_face_t.__index = hb_face_t
+ffi.metatype("hb_face_t", hb_face_t)
+
+-- Dump contents of OT name fields
+function hb_face_t:getNames(maxlen)
```

Comment:
I'm a bit unexperienced with all these metatype and ffi tricks.
But what's this `hb_face_t` ? Where does it end up - as it's not attached to the HB object we get when `require(harfbuzz)` ?
Or is that just attaching additional methods to the libharfbuzz internal hb_face_t type - that other code may use ? (so, these additional methods would be accessible only when used via ffi?)

Reply:
LuaJIT attaches it implicitly to any `hb_face_t*` typed C pointer we spawn through `hb_face_create_*` calls in various places. This is not as unparadigmatic as it may seem - lua does same sort of thing if you attach metatable to string, number or a function - in which case you too get mt per *type*, as opposed to *value* as would happen to UD/table.